catalogue_graph/src/removers/base_graph_remover_incremental.py

```diff
@@ -0,0 +1,88 @@
+from collections.abc import Iterable, Iterator
+from itertools import batched
+
+from utils.aws import get_neptune_client
+from utils.safety import validate_fractional_change
+from utils.types import EntityType
+
+ES_QUERY_NON_VISIBLE_WORKS = {"bool": {"must_not": {"match": {"type": "Visible"}}}}
+
+BATCH_SIZE = 40_000
+
+
+class BaseGraphRemoverIncremental:
```

Comment:
I wonder if we should split this into a base edge & base node remover class, and/or use multiple inheritance (it looks like re-use batching and validation of fractional change are why we handle edge and node removal in the same class). 

Splitting the class might help to keep it clear what the purpose of each class is.